pkg/domain/podman/router.go

```diff
@@ -70,8 +70,9 @@ func (r *RouterEntityManager) QueryConnections(routerId string, edge bool) ([]qd
 		return nil, fmt.Errorf("error querying connections - %w", err)
 	}
 	var connections []qdr.Connection
+	// if the given routerId no longer exists, it is expected to fail
 	err = json.Unmarshal([]byte(data), &connections)
-	if err != nil {
+	if err != nil && routerId == "" {
```

Comment:
Would it be clearer to just check that the routerId is specified as the first line of the function, failing with an error regarding the missing id if not?

Reply:
In this function, the router id is optional. If you do not pass one, then the Unmarshal is expected to work.
If you pass one, it can eventually fail (as skmanage might not return a JSON).

Just to give some context. This function is called to help determine all routers in the VAN (interiors and edges),
so prior to calling this function, it queries the `router.node` entities (interiors), then it iterates through each of 
them, querying their connection entities (which include edges) providing the specific router id to skmanage.

There is a delay between the moment the link is removed and the moment you stop seeing the respective 
`router.node` entity returned by the $management, so although we are still seeing the node entity, it is no 
longer available and that's why it is expected to fail.

Reply:
True stuff. It is really not the place to decide whether or not it is an expected error.
I have delegated this decision to the callers, which are passing a specific router id.